docs/qaoa/binary_paintshop.ipynb

```diff
@@ -0,0 +1,435 @@
+    "    for paint_bitstring in results.measurements[\"z\"]:\n",
+    "        avg_cc += color_changes(paint_bitstring) / repetitions\n",
+    "    return avg_cc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "d2ydfRBrLrnl"
+   },
+   "source": [
+    "We optimize the average number of color changes by adjusting the parameters with scipy.optimzes function minimize. The results of these optimsation runs strongly depend on the random starting values we choose for the parameters, which is why we restart the optimization procedure for different starting parameters 10 times and take the best performing optimized parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "xXPCgWMaSPqJ"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "7.840000000000001"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {
+      "tags": []
+     },
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from scipy.optimize import minimize\n",
+    "\n",
+    "service = cirq.Simulator()\n",
+    "beta, gamma = np.random.rand(2)\n",
+    "average_cc = average_color_changes([beta, gamma])\n",
+    "for _ in range(10):\n",
+    "    initial_guess = np.random.rand(2)\n",
+    "    optimization_result = minimize(\n",
+    "        average_color_changes, initial_guess, method=\"SLSQP\", options={\"eps\": 0.1}\n",
+    "    )\n",
+    "    average_cc_temp = average_color_changes(optimization_result.x)\n",
+    "    if average_cc > average_cc_temp:\n",
+    "        beta, gamma = optimization_result.x\n",
+    "        average_cc = average_cc_temp\n",
+    "average_cc"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "y0tJ2GErNa7w"
+   },
+   "source": [
+    "Note here that the structure of the problem graphs of the binary paintshop problem allow for an alternative technique to come up with good parameters independent of the specifics of the respective instance of the problem: [Training the quantum approximate optimization algorithm without access to a quantum processing unit](https://iopscience.iop.org/article/10.1088/2058-9565/ab8c2b)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "WoBQG2f2L8HC"
+   },
+   "source": [
+    "Once the parameters are optimised, we execute the optimised QAOA circuit 100 times and output the solution with the least color changes.\n",
+    "Please replace `<your key>` with your IonQ API key and `<remote host>` with the API endpoint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "lfUmwcxdo79w"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The minimal number of color changes found by level-1 QAOA is: 6\n",
+      "To achieve this number of color changes the first cars of the car pairs have to be painted with [0 1 0 0 0 0 0 0 0 0]\n"
```

Comment:
This only refers to the first car of the car pair. When expanded to the whole `CAR_SEQUENCE`, this will look like `[0 0 0 0 1 1 0 0 0 0 1 0 0 1 1 1 1 1 1 0]`

Reply:
Aha.  I see that is indeed in the description, but may be worth a second mention for people who zoom through this too fast like me.

Reply:
Added a sentence to elaborate on this.